### PR TITLE
Dev zhi errorhandling

### LINKED ISSUE #4020: Workaround Mono behavior - PackageArchiveReader will throw ArchiveException on mono when package is invalid, nuget doesn't handle it
tracking mono issue

### PR DESCRIPTION
**Pull request by zhili1208**
fixed https://github.com/NuGet/Home/issues/4020

### CODE REVIEW

File: src/NuGet.Core/NuGet.Protocol.Core.v3/Utility/OfflineFeedUtility.cs [hunk 1]
```diff
@@ -207,7 +207,9 @@ public static void ThrowIfInvalid(string path)
                         logger.LogMinimal(message);
                     }
                 }
-                catch (InvalidDataException)
+                // Mono will throw ArchiveException when package is invalid.
+                catch (Exception ex) when( ex is InvalidDataException
```
Comment (joelverhagen) [inline src/NuGet.Core/NuGet.Protocol.Core.v3/Utility/OfflineFeedUtility.cs:211]: `SharpCompress.Common.ArchiveException` == `ex.GetType().FullName`

https://github.com/mono/mono/blob/0bcbe39b148bb498742fc68416f8293ccd350fb6/mcs/class/System.IO.Compression/SharpCompress/Common/ArchiveException.cs

**Review by joelverhagen — APPROVED** [review]
Address comments the :shipit:

File: src/NuGet.Core/NuGet.Protocol.Core.v3/Utility/OfflineFeedUtility.cs (hunk 1 above)
Comment (joelverhagen) [reply src/NuGet.Core/NuGet.Protocol.Core.v3/Utility/OfflineFeedUtility.cs:211]: Check `IsMono`

**Comment by zhili1208** [comment]
https://github.com/NuGet/NuGet.Client/commit/9770cc29557e5b5985c56ab055ee3ec0e99b63df

**Pull request closed without merging**